src/Raven.Server/Utils/ThreadStatUtils.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Raven.Server.Utils
+{
+    public static class ThreadStatUtils
+    {
+        internal static IEnumerable<Thread> GetThreadsOfCurrentProcess()
+        {
+            foreach (Thread t in Process.GetCurrentProcess().Threads)
```

Comment:
`Process.GetCurrentProcess().Threads` is actually returning `ProcessThread`

https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L602

And they already have support for all of the above:

https://github.com/dotnet/corefx/blob/3426b1f983e66f82565cc6a458f7464d1e56a19c/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessThread.Linux.cs#L36

https://github.com/dotnet/corefx/blob/3426b1f983e66f82565cc6a458f7464d1e56a19c/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessThread.Linux.cs#L56

https://github.com/dotnet/corefx/blob/9e6b9c89779f7af21acf2e5817c4f5dde28b25cf/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessThread.Windows.cs#L121